src/orion/client/runner.py

```diff
@@ -0,0 +1,323 @@
+            )
+        except (KeyboardInterrupt, InvalidResult):
+            self.release_all()
+            raise
+
+        to_be_raised = None
+        log.debug(f"Gathered new results {len(results)}")
+
+        # register the results
+        for result in results:
+            self.free_worker += 1
+            trial = self.pending_trials.pop(result.future)
+
+            if isinstance(result, AsyncResult):
+                # NB: observe release the trial already
+                self.client.observe(trial, result.value)
+                self.trials += 1
+
+            if isinstance(result, AsyncException):
+                if (
+                    isinstance(result.exception, ExecutionError)
+                    and result.exception.return_code == self.interrupt_signal_code
+                ):
+                    to_be_raised = KeyboardInterrupt()
+                    self.client.release(trial, status="interrupted")
+                    continue
+
+                if isinstance(result.exception, InvalidResult):
+                    # stop the optimization process if we received `InvalidResult`
+                    # as all the experiments are assumed to be returning those
+                    to_be_raised = result.exception
+
+                # Regular exception, might be caused by the choosen hyperparameters
+                # themselves rather than the code in particular (like Out of Memory error
+                # for big batch sizes)
+                exception = result.exception
+                self.worker_broken_trials += 1
+                self.client.release(trial, status="broken")
+
+                if self.on_error is None or self.on_error(
+                    self, trial, exception, self.worker_broken_trials
+                ):
+                    log.error(result.traceback)
+
+                else:
+                    log.error(str(exception))
+                    log.debug(result.traceback)
+
+                # if we receive too many broken trials, it might indicate the user script
+                # is broken, stop the experiment and let the user investigate
+                if self.worker_broken_trials >= self.max_broken:
+                    raise BrokenExperiment("Worker has reached broken trials threshold")
+
+        if to_be_raised is not None:
+            log.debug("Runner was interrupted")
+            self.release_all()
+            raise to_be_raised
+
+    def release_all(self):
+        """Release all the trials that were reserved by this runner"""
+        # Sanity check
+        for _, trial in self.pending_trials.items():
+            try:
+                self.client.release(trial)
+            except AlreadyReleased:
+                pass
+
+    def _suggest_trials(self, count):
+        """Suggest a bunch of trials to be dispatched to the workers"""
+        trials = []
+        for _ in range(count):
+            try:
+                trial = self.client.suggest()
```

Comment:
Trials should be created in batch by algorithm using `self.client.suggest(pool_size=count)`. It will only return 1 trial, but if the algorithm needs to generate new trials it will do it in batch. We should pass `count` if `self.pool_size == 0` else pass `self.pool_size`.

Reply:
This comment still applies.

Reply:
This is fixed now.

Reply:
yep done